axum/src/extract/path/de.rs

```diff
@@ -79,6 +78,18 @@ impl<'de> Deserializer<'de> for PathDeserializer<'de> {
     parse_single_value!(deserialize_byte_buf, visit_string, "String");
     parse_single_value!(deserialize_char, visit_char, "char");
 
+    fn deserialize_any<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
+        if self.url_params.len() != 1 {
+            return Err(PathDeserializationError::wrong_number_of_parameters()
+                .got(self.url_params.len())
+                .expected(1));
+        }
+        visitor.visit_str(&self.url_params[0].1)
```

Comment:
Hm I'm wondering if this could break something else 🤔 Not very familiar with these kinds of deserializer details.

@jplatte do you know?

Reply:
It doesn't make a lot of sense to me to require there to be exactly one url param for `deserialize_any`. I think deserializing to a free-form thing like `serde_json::Value` should probably get you a sequence of all the values. A sequence of key-value pair like `deserialize_any` on a query string could make sense as well, but here the names are implicit so I'm leaning towards just values.

In any case, no longer erroring in `deserialize_any` isn't going to break anything, but deciding for one behavior of that function means that other reasonable implementations can no longer be added backwards-compatibly.

Overall, I don't think we should merge this. Instead of `serde(flatten)`, `serde(from)` / `serde(try_from)` with an intermediate type (that contains the variants of both inner enums in the example case) would be a cleaner and more efficient way of solving OP's problem. `serde(flatten)` is a kind of ugly thing in general, it's slow and produces very bad error messages when deserialization fails.